src/Shared/Components/DeploymentConfigDiff/DeploymentConfigDiffMain.tsx

```diff
@@ -0,0 +1,154 @@
+import { Fragment, useEffect, useRef, useState } from 'react'
+
+import { ReactComponent as ICSortArrowDown } from '@Icons/ic-sort-arrow-down.svg'
+import { Progressing } from '@Common/Progressing'
+import { CodeEditor } from '@Common/CodeEditor'
+import { MODES, SortingOrder } from '@Common/Constants'
+
+import { SelectPicker } from '../SelectPicker'
+import { DeploymentHistoryDiffView } from '../CICDHistory'
+import { DeploymentConfigDiffAccordion } from './DeploymentConfigDiffAccordion'
+import { DeploymentConfigDiffMainProps, DeploymentConfigDiffSelectPickerProps } from './types'
+
+export const DeploymentConfigDiffMain = ({
+    isLoading,
+    headerText = 'Compare With',
+    configList = [],
+    scrollIntoViewId,
+    selectorsConfig,
+    sortOrder,
+    onSortBtnClick,
+}: DeploymentConfigDiffMainProps) => {
+    // STATES
+    const [expandedView, setExpandedView] = useState<Record<string | number, boolean>>({})
```

Comment:
This state exists in DeploymentConfigDiffNavigation as well. Pls check if redundant

Reply:
2 different states
one for side nav collapsible list and one for accordion in main content